sources/osgUtil/ComposerPostProcess.js

```diff
@@ -931,22 +935,61 @@ utils.createPrototypeObject(
             return source;
         },
 
-        _extractTextures: function(file) {
-            var infos = {
+        setMethodWrapUV: function(method, threshold) {
+            // hook every texture call to handle viewport/texture ratio
+            // when reading out of texture assigned zone
+
+            // 0 - do nothing
+            // 1 - black color when fetching out of bound
+            // 2 - clamp uv to thresold (usually vec2(1.0))
+            if (method !== undefined) this._methodWrapUV = method;
+            if (threshold !== undefined) this._thresholdWrapUV = threshold;
+            this._texInfos = undefined;
+        },
+
+        _getInfos: function() {
+            if (this._texInfos) return this._texInfos;
+
+            var bodySimple = 'texture2D(%tex, (uv) * %ratio)';
+            var bodyNearest =
+                'texture2D(%tex, (floor((uv) * %size) + 0.5) * %ratio / %size, -99999.0)';
+            var bodyBias = 'texture2D(%tex, (uv) * %ratio)';
+
+            // see setMethodWrapUV
+            var val = this._thresholdWrapUV.toExponential();
+            if (this._methodWrapUV === 1) {
+                var prefix = 'step((uv).x, ' + val + ') * step((uv).y, ' + val + ') * ';
+                bodySimple = prefix + bodySimple;
+                bodyNearest = prefix + bodyNearest;
+                bodyBias = prefix + bodyBias;
+            } else if (this._methodWrapUV === 2) {
+                var replaceValue = 'min(uv, vec2(' + val + '))';
+                bodySimple = bodySimple.replace('(uv)', replaceValue);
```

Comment:
strange was sure "replace" call needed a parameter for "global" to be able to replace multiple occurrences ?
```bodySimple.replace(/\(uv\)/g, replaceValue)```

Reply:
There's only 1 occurence of `(uv)` so it's fine.
But I can /g to be safer